viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import { TreeNode, TreeNodeComponent, defaultTreeNode } from './tree-node';
+import { Message } from './message';
+
+type FilterTreeProps = {
+    nodes: TreeNode[];
+    showCheckboxes: boolean;    
+    checkedSeries: number[];                //Optional
+    onChecked: (ids: number[]) => void;     //Optional
+}
+
+type FilterTreeState = {
+    treeNodes: TreeNode[];
```

Comment:
From what  I understand, the part of the TreeNode that is really part of the state of the tree vs the parent is the expanded state, the checked one is managed by the parent, is my understanding correct?